[7.0] quantity warning on lot with negative number

llacroix opened the pull request.

llacroix committed:
[#66647] Init

llacroix committed:
[#66647] Added stock recount

llacroix committed:
[#66647] Added report which control the stock availability and search methods

llacroix committed:
[#66647] Fix PEP8 errors

llacroix committed:
[#66647] Added copyright

llacroix committed:
[#66647] Added doctstrings

llacroix committed:
[#66647] Fix pep8 errors and use orm.Model instead of osv.osv

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

llacroix committed:
[#66647] Fix SQL view

llacroix committed:
[#66647] Add unit of measure on retained stock

llacroix committed:
[#66647] Added translation

llacroix committed:
[#66647] Added tests

gurneyalex committed:
don't use travis container-based infrastructure

llacroix committed:
Add a warning message when there is not enough stock for a split

llacroix committed:
Added French translation

llacroix committed:
Fix move count and added more translatable text

joaoalf commented on quantity_warn_on_lot/__openerp__.py:
```diff
@@ -0,0 +1,45 @@
+        'views/stock_move_split.xml',
+    ],
+    'installable': True,
+    'application': True,
```
@llacroix I don't think that this module qualify as an application.

llacroix replied:
Good point, fixed

joaoalf commented on quantity_warn_on_lot/models/__init__.py:
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
```
@llacroix Copyright?

llacroix replied:
Done

joaoalf commented on quantity_warn_on_lot/models/stock_move_split.py:
```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
```
@llacroix Copyright?

llacroix replied:
Done

joaoalf commented on quantity_warn_on_lot/models/stock_move_split.py:
```diff
@@ -0,0 +1,185 @@
+        """
+        base_func = super(StockMoveSplit, self).split_lot
+
+        try:
```
@llacroix Why not use a orm_except and let odoo handle the roolback?

llacroix replied:
It's not an orm exception. I had in mind to do that at first but this will cause other problems. If a real orm exception is raised. It will get caught there and it's possible that the other exceptions will get silenced by a weird wizard that pops up.

In this case, we want to handle the exception without raising errors.

joaoalf replied:
@llacroix I'm ok with the exception part. The thing is that resort to manual transaction control usually means that the developper is trying to accomplish something in a non-odoo way. maybe what you need is a validation step (a 'verify stock' button or something like that) to check if the split can be performed before trying to perform it.

llacroix replied:
@joaoalf I understand. The goal of the module is to add a check before the split. Which means that the user has to click on the split and if something wrong happens, it will show the popup. Otherwise it does as it always did. Though, as you say, it seems a bit strange and the wizard part isn't very elegant though. The other problem is that it is tracking which button are pressed.

A validation button wouldn't speak as much as a dialog box saying "You're doing something wrong. But you can keep going with it or correct it." That said, the check is just a warning and doesn't override the real validation that the split button does.

joaoalf commented on quantity_warn_on_lot/models/stock_move_split.py:
```diff
@@ -0,0 +1,185 @@
+        self.quantity = quantity
+
+
+class StockMoveSplit(orm.TransientModel):
```
@llacroix In version 7.0 is more common to put wizards (models and views) in the wizard namespace (folder).

llacroix replied:
I'll try to move that in a different folder with the according views.

llacroix committed:
Fix pep8 error

llacroix committed:
Added copyrights

llacroix committed:
Catch value error in case view isn't yet loaded

joaoalf commented on quantity_warn_on_lot/__init__.py:
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
+
```
Copyright?

llacroix committed:
Add copyrights

llacroix committed:
Adding xml version

ehdem committed:
[UPD] Remove view and clean the model

No need adding the view and just overriding
one method in the StockMoveSplit class

joaoalf commented on quantity_warn_on_lot/models/stock_move_split.py:
```diff
@@ -0,0 +1,73 @@
+        product_obj = self.pool.get('product.product')
+        for data in self.browse(cr, uid, ids, context=context):
+            for move in move_obj.browse(cr, uid, move_ids, context=context):
+                product_uom = product_obj.browse(
```
@ehdem Why not `move.product_uom` ?

joaoalf commented on quantity_warn_on_lot/models/stock_move_split.py:
```diff
@@ -0,0 +1,71 @@
+##############################################################################
+import logging
+
+from openerp.osv import orm, osv
```
@ehdem Why import osv?

joaoalf commented on quantity_warn_on_lot/models/stock_move_split.py:
```diff
@@ -0,0 +1,71 @@
+                        product_uom, context=context)
+                    if (data.location_id.usage == 'internal') and \
+                            (quantity > (amount_actual or 0.0)):
+                        raise osv.except_osv(
```
@ehdem Why not `raise orm.except_orm()`?

yvaucher commented:
To rebase

gurneyalex commented on .travis.yml:
```diff
@@ -1,3 +1,4 @@
+sudo: required
```
to remove.

Cedric-Pigeon commented:
@llacroix Hello, do you plan to achieve this PR ? Is it still usefull ? Thanks.

llacroix commented:
@Cedric-Pigeon not planning anything. Can't say if it's useful. Though not for me.
